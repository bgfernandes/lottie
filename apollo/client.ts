
import { ApolloClient, InMemoryCache, NormalizedCacheObject } from '@apollo/client'

let apolloClient: ApolloClient<NormalizedCacheObject>

if (typeof window === 'undefined') {
  // code is running in the server side
  const { SchemaLink } = require('@apollo/client/link/schema')
  const { schema } = require('./schema')

  apolloClient = new ApolloClient({
    ssrMode: true,
    link: new SchemaLink({ schema }),
    cache: new InMemoryCache(),
  })
} else {
  // code is running in the browser
  const { HttpLink } = require('@apollo/client/link/http')

  apolloClient = new ApolloClient({
    ssrMode: false,
    link: new HttpLink({
      uri: '/api/graphql',
      credentials: 'same-origin',
    }),
    cache: new InMemoryCache(),
  })
}

export default apolloClient
