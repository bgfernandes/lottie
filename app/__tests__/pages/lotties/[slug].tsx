/**
 * @jest-environment jsdom
 */

 import { render } from '@testing-library/react'
 import { MockedProvider } from '@apollo/client/testing'
import ShowLottieFilePage from '../../../pages/lotties/[slug]'

 describe('Share', () => {
   it('renders successfully when there is a lottie', () => {
     const lottie = {
      id: '4',
      slug: 'IDxl4QWcIVEWAtVTcF6ID',
      url: 'http://localhost:3001/local_file_store/IDxl4QWcIVEWAtVTcF6ID.json',
      createdAt: '1630786408304',
      updatedAt: '1630786408304'
    }

     const { container } = render(
       <MockedProvider mocks={[]} addTypename={false}>
         <ShowLottieFilePage lottieFile={lottie} />
       </MockedProvider>
     )

     expect(container).toMatchSnapshot()
   })

   it('renders successfully when the lottie is not found', () => {
    const { container } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <ShowLottieFilePage lottieFile={null} />
      </MockedProvider>
    )

    expect(container).toMatchSnapshot()
   })
 })
