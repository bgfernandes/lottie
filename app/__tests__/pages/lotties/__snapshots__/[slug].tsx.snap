// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share renders successfully when the lottie is not found 1`] = `
<div>
  <div>
    Lottie not found.
  </div>
</div>
`;

exports[`Share renders successfully when there is a lottie 1`] = `
<div>
  <div>
    Slug: 
    IDxl4QWcIVEWAtVTcF6ID
    <br />
    Url: 
    http://localhost:3001/local_file_store/IDxl4QWcIVEWAtVTcF6ID.json
    <br />
    Created At: 
    Thu Jul 25 53647 16:31:44 GMT+0000
    <br />
    Last Updated At: 
    Thu Jul 25 53647 16:31:44 GMT+0000
  </div>
</div>
`;
